tests/deprecated_api/test_remove_1-5.py

```diff
@@ -196,7 +196,7 @@ def __init__(self):
             self.automatic_optimization = False
 
         def training_step(self, batch, batch_idx, optimizer_idx):
-            assert optimizer_idx is not None
+            assert optimizer_idx is None
```

Comment:
why has this changed?

Reply:
Here, before, optimizer_idx=0 which only exists for backward compatibility when someone has the optimizer_idx argument in the signature of training_step. The value is arbitrary and has no meaning in manual optimization.

Why did it change?
In the above refactor, I grouped the optimization step under one function which is called either

`self.run_optimization(batch_idx, split_idx, split_batch, opt_idx, optimizer)`
in auto optimization 

or 

`self.run_optimization(batch_idx, split_idx, split_batch, opt_idx=None, optimizer=None)`
in manual optimization.

I can change it back to `opt_idx=0` but I find that suboptimal, since referring to the first optimizer is wrong.

Reply:
Actually I would change It back, since this means the same code won't run when you switch from 1 to to 2 optims if you check that value ( I get your point though, but also when you only have one optimal isn't it always the first (and only)?)

Reply:
It would anyway not run, because we don't iterate over optimizers in manual optimization and the optimizer_idx argument is an artifact from deprecating that argument from training_step. 

If the user wants multiple optimizers, they have to do:

```python
opt1, opt2 = self.optimizers()

self.manual_backward(loss1)
opt1.step()
self.manual_backward(loss2)
opt2.step()
```

Otherwise nothing works in manual optimization.